File: backend/app/schemas/analytics.py
"""Analytics response schemas."""
from __future__ import annotations
from pydantic import BaseModel
from typing import Optional


class AnalyticsSummary(BaseModel):
    total_cost_usd: float
    total_requests: int
    total_tokens: int
    active_agents: int
    avg_cost_per_request: float
    error_rate_pct: float


class TimeseriesPoint(BaseModel):
    timestamp: str
    cost_usd: float
    requests: int


class AgentBreakdown(BaseModel):
    agent_id: str
    agent_name: str
    cost_usd: float
    pct: float


class ProviderBreakdown(BaseModel):
    provider: str
    cost_usd: float
    pct: float


class ModelBreakdown(BaseModel):
    model: str
    provider: str
    cost_usd: float
    pct: float


class TeamBreakdown(BaseModel):
    team_label: str
    cost_usd: float
    pct: float


class AnalyticsResponse(BaseModel):
    summary: AnalyticsSummary
    timeseries: list[TimeseriesPoint]
    by_agent: list[AgentBreakdown]
    by_provider: list[ProviderBreakdown]
    by_model: list[ModelBreakdown]
    by_team: list[TeamBreakdown]

File: frontend/src/types/analytics.ts
export interface AnalyticsSummary {
  total_cost_usd: number
  total_requests: number
  total_tokens: number
  active_agents: number
  avg_cost_per_request: number
  error_rate_pct: number
}

export interface TimeseriesPoint {
  timestamp: string
  cost_usd: number
  requests: number
}

export interface AgentBreakdown {
  agent_id: string
  agent_name: string
  cost_usd: number
  pct: number
}

export interface ProviderBreakdown {
  provider: string
  cost_usd: number
  pct: number
}

export interface ModelBreakdown {
  model: string
  provider: string
  cost_usd: number
  pct: number
}

export interface TeamBreakdown {
  team_label: string
  cost_usd: number
  pct: number
}

export interface AnalyticsResponse {
  summary: AnalyticsSummary
  timeseries: TimeseriesPoint[]
  by_agent: AgentBreakdown[]
  by_provider: ProviderBreakdown[]
  by_model: ModelBreakdown[]
  by_team: TeamBreakdown[]
}

// Live event from WebSocket
export interface LiveEvent {
  event_id: string
  agent: string
  model: string
  cost_usd: number
  status: string
  tracked_at: string
}
